drivers/gpu/drm/amd/display/dc/basics/dce_calcs.c

```diff
@@ -3045,8 +3045,8 @@ bool bw_calcs(struct dc_context *ctx,
 	int pipe_count,
 	struct dce_bw_output *calcs_output)
 {
-	struct bw_calcs_data *data = kzalloc(sizeof(struct bw_calcs_data),
-					     GFP_KERNEL);
+	struct bw_calcs_data *data = kvzalloc(sizeof(struct bw_calcs_data),
+					      GFP_KERNEL);
```

Comment:
Could you please keep the Linux code unchanged and have the FreeBSD specific version separately?

```c
#ifdef __FreeBSD__
/* Your patched code */
#else
/* Original code */
#endif
```

Reply:
Done.